src/middlewared/middlewared/plugins/cloud_backup/restore.py

```diff
@@ -0,0 +1,44 @@
+from middlewared.plugins.cloud_backup.restic import get_restic_config, run_restic
+from middlewared.schema import accepts, Dict, Int, List, Str
+from middlewared.service import job, Service
+from middlewared.validators import NotMatch
+
+
+class CloudBackupService(Service):
+
+    class Config:
+        cli_namespace = "task.cloud_backup"
+        namespace = "cloud_backup"
+
+    @accepts(
+        Int("id"),
+        Str("snapshot_id", validators=[NotMatch(r"^-")]),
+        Str("subfolder"),
+        Str("destination_path", validators=[NotMatch(r"^-")]),
```

Comment:
Are these local filesystem paths? Do we need to prevent people from doing silly things like overwriting `/etc`?